### PR TITLE
Logging middleware for testkit

### PR DESCRIPTION
**Pull request by slowli**
As a part of bringing #459 to shape, this PR introduces logging middleware for the testkit API and adds corresponding tests.

### CODE REVIEW

File: testkit/src/lib.rs [hunk 1]
```diff
@@ -280,7 +281,7 @@ impl TestKitBuilder {
     /// Creates the testkit.
     pub fn create(self) -> TestKit {
         if self.logger {
-            exonum::helpers::init_logger().unwrap();
+            exonum::helpers::init_logger().ok();
```
Comment (vitvakatu) [inline testkit/src/lib.rs]: What's the purpose of this change?

File: testkit/src/api.rs [hunk 2]
```diff
@@ -311,6 +307,96 @@ impl TestKitApi {
...
+            .unwrap_or_default(),
+        resp_status = response
+            .status
+            .as_ref()
+            .map(ToString::to_string)
+            .unwrap_or_else(|| "(no status)".to_string()),
+        resp_body = response_body
+            .as_ref()
+            .map(String::as_ref)
+            .unwrap_or_else(|| "(no body)")
```
Comment (vitvakatu) [inline testkit/src/api.rs]: `unwrap_or`?

File: testkit/src/api.rs [hunk 3]
```diff
@@ -311,6 +307,96 @@ impl TestKitApi {
...
+        method = request.method,
+        access = access,
+        url = url,
+        req_body = req_body
+            .map(|body| format!("\nBody: {}", body))
+            .unwrap_or_default(),
+        resp_status = response
+            .status
+            .as_ref()
+            .map(ToString::to_string)
```
Comment (vitvakatu) [inline testkit/src/api.rs]: Why not using `String::as_ref` here too?

File: testkit/src/api.rs [hunk 4]
```diff
@@ -311,6 +307,96 @@ impl TestKitApi {
...
+        })
+    };
+
+    trace!(
+        "{method} ({access}) /{url}{req_body}\nResponse: {resp_status}\n{resp_body}\n",
+        method = request.method,
+        access = access,
+        url = url,
+        req_body = req_body
+            .map(|body| format!("\nBody: {}", body))
```
Comment (vitvakatu) [inline testkit/src/api.rs]: nitpick: I'm not a fan of avoiding **all** allocations in code, but you can easily avoid any of them in this part of the code (by the price of less readable message in case of no request body).

File: testkit/src/lib.rs (hunk 1 above)
Comment (slowli) [reply testkit/src/lib.rs]: There is a separate commit dedicated to it :slightly_smiling_face: Basically, if you try to enable the logger in an integration test (say, `counter`) via

```rust
fn init_testkit() -> (TestKit, TestKitApi) {
    let testkit = TestKitBuilder::validator()
        .with_service(CounterService)
        .with_logger()
        .create();
    // ...
}
```

then all tests except the first one will panic on this `unwrap`.

File: testkit/src/api.rs (hunk 3 above)
Comment (slowli) [reply testkit/src/api.rs]: `response.status` has type `Option<Status>`, so `ToString::to_string` converts it to the string form. If I understand correctly, in order to use `String::as_ref`, you need to map `Option<Status>` to `Option<String>` beforehand, which is sort of clunky.
